### PR TITLE
Third Week PR - Serving

### PR DESCRIPTION
**Pull request by benitomartin**
## Third PR - Branch: serving

- Reorganized Notebooks Folders
- Changed env var "CONFIG DATABRICKS" and "CODE_PATH"
- New Workspace mlops_students (change catalog and schema name)
- Pyarrow incompatibility with mlflow/feature lookup. Changed to 14.0.2 in wheel 0.0.9
- Added Notebooks feature/model serving
- Config file: added A/B test parameters

## Release Notes

- **New Features**
  - Introduced A/B testing framework for credit default prediction.
  - Added functionality for creating and serving feature tables in Databricks.
  - Implemented model serving endpoints for real-time predictions.

- **Improvements**
  - Enhanced error handling and logging in data preprocessing and cleaning processes.
  - Updated configuration files to reflect new catalog and schema names.
  - Improved README documentation to clarify recent changes and new features.

- **Bug Fixes**
  - Resolved compatibility issues with `pyarrow` and `mlflow`.

- **Chores**
  - Updated project dependencies and versioning in `pyproject.toml`.

### CODE REVIEW

File: notebooks/model_feature_serving/model_serving_notebook.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+from credit_default.utils import load_config
+
+# COMMAND ----------
+
+workspace = WorkspaceClient()
+spark = SparkSession.builder.getOrCreate()
+
+# COMMAND ----------
+
+config = load_config("/Volumes/mlops_students/benitomartin/config/project_config.yml")
```
Comment (mvechtomova) [inline notebooks/model_feature_serving/model_serving_notebook.py:19]: not sure how handy it is to have config file in the volume

File: notebooks/model_feature_serving/model_serving_notebook.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+token = dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiToken().get()  # noqa: F821
+host = spark.conf.get("spark.databricks.workspaceUrl")
+
+# COMMAND ----------
+
+print(host)
+
+# COMMAND ----------
+
+required_columns = [
```
Comment (mvechtomova) [inline notebooks/model_feature_serving/model_serving_notebook.py:112]: maybe nice to have it somewhere in the config file

**Review by mvechtomova — APPROVED** [review]
Nice work! Approved

File: notebooks/feature_engineering/basic_mlflow_experiment_notebook.py [hunk 3]
```diff
@@ -22,13 +22,16 @@
 CONFIG_DATABRICKS = os.environ["CONFIG_DATABRICKS"]
 PROFILE = os.environ["PROFILE"]
 print(CONFIG_DATABRICKS)
+print(PROFILE)
 
 # COMMAND ----------
+
 # tracking and registry URIs
 mlflow.set_tracking_uri(f"databricks://{PROFILE}")
```
[the remark this answers is not in the document]
Comment (benitomartin) [reply notebooks/feature_engineering/basic_mlflow_experiment_notebook.py:30]: This notebook is from the previous PR. I had issues to run it without profile, but I solved in the new notebooks. Will change it

File: notebooks/model_feature_serving/model_serving_notebook.py (hunk 1 above)
Comment (benitomartin) [reply notebooks/model_feature_serving/model_serving_notebook.py:19]: I will change it to repo folder

File: notebooks/model_feature_serving/model_serving_notebook.py (hunk 2 above)
Comment (benitomartin) [reply notebooks/model_feature_serving/model_serving_notebook.py:112]: Yes, I can move it there

**Pull request merged**